indico/modules/events/registration/controllers/display.py

```diff
@@ -0,0 +1,115 @@
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request, session, redirect, flash
+from werkzeug.exceptions import Forbidden
+
+from indico.core.db import db
+from indico.modules.auth.util import redirect_to_login
+from indico.modules.events.registration.models.registration_forms import RegistrationForm
+from indico.modules.events.registration.models.registrations import Registration
+from indico.modules.events.registration.util import (get_event_section_data, make_registration_form,
+                                                     was_regform_submitted, save_registration_to_session)
+from indico.modules.events.registration.views import (WPDisplayRegistrationFormConference,
+                                                      WPDisplayRegistrationFormMeeting,
+                                                      WPDisplayRegistrationFormLecture)
+from indico.modules.payment import event_settings
+from indico.util.i18n import _
+from indico.web.flask.util import url_for
+from MaKaC.webinterface.rh.conferenceDisplay import RHConferenceBaseDisplay
+
+
+def _can_redirect_to_single_regform(regforms):
+    return len(regforms) == 1 and regforms[0].can_submit(session.user) and not was_regform_submitted(regforms[0])
+
+
+class RHRegistrationFormDisplayBase(RHConferenceBaseDisplay):
+
+    CSRF_ENABLED = True
+
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.event = self._conf
+
+    @property
+    def view_class(self):
+        mapping = {
+            'conference': WPDisplayRegistrationFormConference,
+            'meeting': WPDisplayRegistrationFormMeeting,
+            'simple_event': WPDisplayRegistrationFormLecture
+        }
+        return mapping[self.event.getType()]
+
+
+class RHRegistrationFormList(RHRegistrationFormDisplayBase):
+    """List of all registration forms in the event"""
+
+    def _process(self):
+        regforms = RegistrationForm.find_all(RegistrationForm.is_visible, event_id=int(self.event.id))
+        if _can_redirect_to_single_regform(regforms):
+            return redirect(url_for('.display_regform', regforms[0]))
+        return self.view_class.render_template('display/regforms_list.html', self.event, regforms=regforms,
+                                               event=self.event, was_regform_submitted=was_regform_submitted)
+
+
+class RHRegistrationFormSubmit(RHRegistrationFormDisplayBase):
```

Comment:
Is this really a submit `RH`? It looks to me it's also displaying the registration form.